chapters/operatorsandexpressions.tex

```diff
@@ -1379,6 +1379,35 @@ \section{Variability of Expressions}\label{variability-of-expressions}
   discrete-time expression.
 \end{itemize}
 
+\begin{example}
+The (underdetermined) model \lstinline!Test! below illustrates two kinds of consequences due to variability constraints.
+First, it contains variability errors for declaration equations and assignments.
+Second, it illustrates the impact of variability on the matching of equations to variables, which can lead to violation of the perfect matching rule.
+Details of how variabilities are determined are given in the sections below.
+\begin{lstlisting}[language=modelica]
+model Constants
+  parameter Real p1 = 1;
+  constant Real c1 = p1 + 2; // error, not a constant expression
+  parameter Real p2 = p1 + 2; // fine
+end Constants;
+model Test
+  Constants c1(p1=3); // fine
+  Constants c2(p2=7); // fine, declaration equation can be modified
+  Real x;
+  Boolean b1 = noEvent(x > 1); // error, since b1 is a discrete-time variable
+                               // and noEvent(x > 1) is not discrete-time.
+  Boolean b2;
+  Integer i1;
+  Integer i2;
+algorithm
+  i1 := x; // error, assignment to variable of lesser variability.
+equation
+  b2 = noEvent(x > 1); // no variability error, but equation cannot be matched.
```

Comment:
This should now reference the rule on line 1501, and not just mention matching.

Reply:
OK, fixed.  This equations can be rejected for more than one reason, and the rule is now referenced by giving it the name the _discrete-valued equation variability rule_.